### DISCUSSION TITLE
uninitialized constant Pundit::Authorization

### DISCUSSION THREAD

**Discussion by albertski** [varvet/pundit#799] [Q & A] [answered]
Below is a simplified version of a concern I use in my Controllers.

```
module One
  module Two
    module PunditAuthorization
      extend ActiveSupport::Concern
      include Pundit::Authorization
    end
  end
end
```
When I upgraded to Rails 7, I got the following error:

```
my-engine/controllers/concerns/one/two/pundit_authorization.rb:6:in `<module:PunditAuthorization>': uninitialized constant One::Two::PunditAuthorization::Pundit (NameError)

      include Pundit::Authorization
              ^^^^^^^^^^^^^^^
```

The issue disappears if I use `include Pundit` instead of `include Pundit::Authorization`.  Should we be using `include Pundit::Authorization` in Rails 7?

**Comment by Burgestrand** [comment]
No, `include Pundit` is deprecated and will be removed in a future release.

With the limited information provided my first guess is that you might be using an old version of Pundit?

**Comment by albertski** [comment]
Thanks. I am using the latest ([v2.3.1](https://github.com/varvet/pundit/releases/tag/v2.3.1)).  The pundit gem is included in one of our engine gems so let me keep digging to see if something isn't set up correctly on our end.

**Comment by Burgestrand** [comment]
I didn't notice the hint in the error, but there's one:
> uninitialized constant One::Two::PunditAuthorization::Pundit

I'm not sure why `include Pundit::Authorization` would be different from `include Pundit` in your case; but because of this error message I'm guessing `include ::Pundit::Authorization` would work here (note the prefix `::`).

**Comment by albertski** [comment, marked as answer]
Sorry for the late response @Burgestrand and thanks for the suggestion.

Turns out I wasn't using the latest version of Pundit. I had multiple gems requiring pundit and one of them was still on an older version.  I have it working with `include Pundit::Authorization`.

**Comment by Burgestrand** [comment]
Thanks for replying, that's good news!
